pkg/controller/compliancescan/compliancescan_controller.go

```diff
@@ -658,19 +658,21 @@ func (r *ReconcileComplianceScan) phaseAggregatingHandler(h scanTypeHandler, log
 		return reconcile.Result{}, err
 	}
 
-	instanceCopy := instance.DeepCopy()
-
-	if instanceCopy.Annotations == nil {
-		instanceCopy.Annotations = make(map[string]string)
+	// check if instance has any annotation before checking for check count
+	// to avoid nil pointer exception in case of missing annotation
+	if instance.Annotations == nil {
+		instance.Annotations = make(map[string]string)
 	}
-
-	// adding check count annotation
-	instanceCopy.Annotations[compv1alpha1.ComplianceCheckCountAnnotation] = strconv.Itoa(checkCount)
-
-	if err := r.Client.Update(context.TODO(), instanceCopy); err != nil {
-		logger.Error(err, "Cannot update the scan with the check count")
-		return reconcile.Result{}, err
+	if _, ok := instance.Annotations[compv1alpha1.ComplianceCheckCountAnnotation]; !ok {
```

Comment:
Do we need to perform this conditional check if we've already created the annotations on the instance if they didn't exist?

Reply:
Nevermind - this is still needed because we're checking for a specific annotation and not the presences of all annotations.